packages/core/src/dialects/abstract/query-generator-typescript.ts

```diff
@@ -742,20 +762,147 @@ export class AbstractQueryGeneratorTypeScript {
-      this.#internals.addLimitAndOffset(options),
+      `DELETE FROM ${table}`,
+      whereOptions.where ? this.whereQuery(whereOptions.where, whereOptions) : '',
+      this.#internals.addLimitAndOffset(whereOptions),
     ]);
   }
 
+  /**
+   * Returns an update query
+   *
+   * @param tableOrModel
+   * @param attrValueHash
+   * @param options
+   */
+  updateQuery(
+    tableOrModel: TableOrModel,
+    attrValueHash: Record<string, unknown>,
+    options?: UpdateQueryOptions,
+  ): QueryWithBindParams {
+    if (options) {
+      rejectInvalidOptions(
+        'updateQuery',
+        this.dialect,
+        UPDATE_QUERY_SUPPORTABLE_OPTIONS,
+        {
+          ...this.dialect.supports.update,
+          limit: this.dialect.supports.update.limit || this.dialect.supports.update.modelWithLimit,
+        },
+        options,
+      );
+    }
+
+    if (typeof attrValueHash !== 'object' || attrValueHash == null || Array.isArray(attrValueHash)) {
+      throw new Error(`Invalid value: ${NodeUtil.inspect(attrValueHash)}. Expected an object.`);
+    }
+
+    const bind = Object.create(null);
+    const valueHash = removeNullishValuesFromHash(attrValueHash, this.options.omitNull ?? false);
+    const attributeMap = new Map<string, NormalizedDataType>();
+    const modelDefinition = extractModelDefinition(tableOrModel);
+    const updateOptions: UpdateQueryOptions = {
+      ...options,
+      model: modelDefinition,
+      bindParam: options?.bindParam === undefined ? createBindParamGenerator(bind) : options.bindParam,
+    };
+
+    if (this.sequelize.options.dialectOptions.prependSearchPath || updateOptions.searchPath) {
+      // Not currently supported with search path (requires output of multiple queries)
+      updateOptions.bindParam = undefined;
+    }
+
+    if (modelDefinition) {
+      if (updateOptions.columnTypes) {
+        throw new Error('Using options.columnTypes in updateQuery requires specifying a model or model definition.');
+      }
+
+      for (const [column, attribute] of modelDefinition.physicalAttributes) {
+        attributeMap.set(attribute.columnName ?? column, attribute.type);
+      }
+    } else if (options?.columnTypes) {
+      for (const column of Object.keys(options.columnTypes)) {
+        attributeMap.set(column, this.sequelize.normalizeDataType(options.columnTypes[column]));
+      }
+    }
+
+    const updateFragment: string[] = [];
+    for (const column of Object.keys(valueHash)) {
+      const rowValue = valueHash[column];
+      if (modelDefinition?.columns.get(column)?.autoIncrement && !this.dialect.supports.autoIncrement.update) {
+        // Skip auto-increment fields if the dialect does not support updating them
+        throw new Error(`The ${this.dialect.name} dialect does not support updating auto-increment fields.`);
+      } else if (rowValue === undefined) {
+        // Treat undefined values as non-existent
+        continue;
+      }
```

Comment:
as we treat undefined and missing properties equally, setting an autoIncrement column to undefined should not throw

```suggestion
      if (rowValue === undefined) {
        // Treat undefined values as non-existent
        continue;
      }
      
      if (modelDefinition?.columns.get(column)?.autoIncrement && !this.dialect.supports.autoIncrement.update) {
        // Skip auto-increment fields if the dialect does not support updating them
        throw new Error(`The ${this.dialect.name} dialect does not support updating auto-increment fields.`);
      }
```

Reply:
Updated now.